pkg/document/json/rga_tree_split.go

```diff
@@ -222,10 +222,22 @@ func (t *RGATreeSplitNode) Value() RGATreeSplitValue {
 	return t.value
 }
 
+// removedNodesMapKey is a key of removedNodesMap.
+// The shape of the key is '{createdAt.Key()}:{offset}'.
+type removedNodesMapKey string
+
+// createRemovedNodesMapKey generates keys for removedNodes.
+func createRemovedNodesMapKey(createdAtKey string, offset int) removedNodesMapKey {
+	key := fmt.Sprintf("%s:%d", createdAtKey, offset)
+	return removedNodesMapKey(key)
+}
+
 type RGATreeSplit struct {
 	initialHead *RGATreeSplitNode
 	treeByIndex *splay.Tree
 	treeByID    *llrb.Tree
+
+	removedNodesMap map[removedNodesMapKey]*RGATreeSplitNode
```

Comment:
Since `map` already has a plural meaning, it would be nice to use the `{singular}Map` pattern to name this map(`removedNodesMap` -> `removedNodeMap`).